### PR TITLE
Search feature

### PR DESCRIPTION
**Pull request by ebrahim-elgaml**
Add search feature to search images by name

### CODE REVIEW

File: app/controllers/history.js [hunk 1]
```diff
@@ -2,8 +2,10 @@ import Ember from 'ember';
 
 export default Ember.Controller.extend({
 
-  sortProperties: ['name:desc'],
-  sortedModel: Ember.computed.sort('model', 'sortProperties'),
+  queryChanged: function() {
+    this.set('filteredModel', this.get('model').filter( item => item.name.toLowerCase().includes(this.get('q'))).sort((a, b) => a <= b ? 1 : -1));
+  }.observes('q', 'model'),
```
Comment (galfert) [inline app/controllers/history.js]: In a lot of cases, using an observer is considered bad practice.
Here, you can just define `filteredModel` to be a computed property that depends on the keys `q` and `model` instead.

**Comment by ebrahim-elgaml** [comment]
@galfert okay I will modify it but could you please explain to me why it is considered as a bad practice?

**Comment by ebrahim-elgaml** [comment]
@galfert Please check again

**Comment by ebrahim-elgaml** [comment]
@galfert
Thank you very much I will check the video.
Can you review the changes and tell me if the approach I have followed is the correct one or not

File: app/controllers/history.js [hunk 2]
```diff
@@ -3,8 +3,12 @@ import Ember from 'ember';
 export default Ember.Controller.extend({
 
   sortProperties: ['name:desc'],
+  queryParams: ['q'],
+  q: '',
   sortedModel: Ember.computed.sort('model', 'sortProperties'),
-
+  filteredModel: Ember.computed.filter('sortedModel', function(item) {
+    return item.name.toLowerCase().includes(this.get('q'));
```
Comment (galfert) [inline app/controllers/history.js]: If I use any capitalization in the search term, it won't find anything. You might want to  convert `q` to lowercase as well.

File: app/templates/history.hbs [hunk 3]
```diff
@@ -1,6 +1,9 @@
 <section class="content history">
+  <div class="search">
+    {{input value=q placeholder="Search" class="search-field"}}
```
Comment (galfert) [inline app/templates/history.hbs]: Why not use `type="search"` here?

File: app/controllers/history.js (hunk 2 above)
Comment (ebrahim-elgaml) [inline app/controllers/history.js]: Yes you are right. I will modify it

File: app/templates/history.hbs (hunk 3 above)
Comment (ebrahim-elgaml) [inline app/templates/history.hbs]: I did not know about it but I will try it

**Comment by ebrahim-elgaml** [comment]
@galfert I have modified both. You can check again now

**Comment by galfert** [comment]
The code looks good to me now 👍

**Comment by ebrahim-elgaml** [comment]
Okay thanks @galfert

**Comment by skddc** [comment]
I rebased this on master, refactored it a bit, and improved the styles for both mobile and desktop screens.

Thanks, @ebrahim-elgaml!

**Pull request closed without merging**